homeassistant/components/channels/media_player.py

```diff
@@ -113,106 +100,43 @@ def update_favorite_channels(self):
     def update_state(self, state_hash):
         """Update all the state properties with the passed in dictionary."""
         self.status = state_hash.get("status", "stopped")
-        self.muted = state_hash.get("muted", False)
-
+        self._attr_is_volume_muted = state_hash.get("muted", False)
         channel_hash = state_hash.get("channel")
         np_hash = state_hash.get("now_playing")
-
+        channel_image_url = None
+        self._attr_media_content_id = None
         if channel_hash:
-            self.channel_number = channel_hash.get("channel_number")
-            self.channel_name = channel_hash.get("channel_name")
-            self.channel_image_url = channel_hash.get("channel_image_url")
-        else:
-            self.channel_number = None
-            self.channel_name = None
-            self.channel_image_url = None
-
+            channel_image_url = channel_hash.get("channel_image_url")
+            self._attr_media_content_id = channel_hash.get("channel_number")
+        now_playing_title = now_playing_image_url = None
         if np_hash:
-            self.now_playing_title = np_hash.get("title")
-            self.now_playing_episode_title = np_hash.get("episode_title")
-            self.now_playing_season_number = np_hash.get("season_number")
-            self.now_playing_episode_number = np_hash.get("episode_number")
-            self.now_playing_summary = np_hash.get("summary")
-            self.now_playing_image_url = np_hash.get("image_url")
-        else:
-            self.now_playing_title = None
-            self.now_playing_episode_title = None
-            self.now_playing_season_number = None
-            self.now_playing_episode_number = None
-            self.now_playing_summary = None
-            self.now_playing_image_url = None
-
-    @property
-    def name(self):
-        """Return the name of the player."""
-        return self._name
-
-    @property
-    def state(self):
-        """Return the state of the player."""
+            now_playing_title = np_hash.get("title")
+            now_playing_image_url = np_hash.get("image_url")
+        self._attr_source_list = [channel["name"] for channel in self.favorite_channels]
+        self._attr_media_title = now_playing_title if self._attr_state else None
+        self._attr_media_image_url = (
+            now_playing_image_url
+            if now_playing_image_url
+            else channel_image_url
+            if channel_image_url
+            else "https://getchannels.com/assets/img/icon-1024.png"
+        )
```

Comment:
This is really hard to read, the original logic makes more sense to me. These attributes are created to allow for making things simpler; this has the opposite effect.